cloudplatform/connectivity-destination-service/src/main/java/com/sap/cloud/sdk/cloudplatform/connectivity/TransparentProxyDestination.java

```diff
@@ -0,0 +1,671 @@
+            }
+
+            if( tenant instanceof DefaultTenant defaultTenant
+                && !containsHeader(allHeaders, TENANT_SUBDOMAIN_HEADER_KEY)
+                && !containsHeader(allHeaders, TENANT_ID_HEADER_KEY)
+                && defaultTenant.getSubdomain() != null
+                && !defaultTenant.getSubdomain().isEmpty() ) {
+                allHeaders.add(new Header(TENANT_SUBDOMAIN_HEADER_KEY, defaultTenant.getSubdomain()));
+            }
+        });
+
+        AuthTokenAccessor.tryGetCurrentToken().onSuccess(token -> {
+            if( !containsHeader(allHeaders, AUTHORIZATION_HEADER_KEY)
+                && token.getJwt().getToken() != null
+                && !token.getJwt().getToken().isEmpty() ) {
+                allHeaders.add(new Header(AUTHORIZATION_HEADER_KEY, token.getJwt().getToken()));
+            }
+        });
+
+        return allHeaders;
+    }
+
+    boolean containsHeader( final Collection<Header> headers, final String headerName )
+    {
+        return headers.stream().anyMatch(h -> h.getName().equalsIgnoreCase(headerName));
+    }
+
+    private List<Header> getHeadersFromHeaderProviders( @Nonnull final URI requestUri )
+    {
+        final List<DestinationHeaderProvider> aggregatedHeaderProviders = new ArrayList<>();
+        aggregatedHeaderProviders.addAll(customHeaderProviders);
+        aggregatedHeaderProviders.addAll(headerProvidersFromClassLoading);
+
+        final String msg = "Found these {} destination header providers: {}";
+        log.debug(msg, aggregatedHeaderProviders.size(), aggregatedHeaderProviders);
+
+        final DestinationRequestContext requestContext = new DestinationRequestContext(this, requestUri);
+
+        final List<Header> result = new ArrayList<>();
+        for( final DestinationHeaderProvider headerProvider : aggregatedHeaderProviders ) {
+            try {
+                result.addAll(headerProvider.getHeaders(requestContext));
+            }
+            catch( final Exception e ) {
+                final String err =
+                    "Header provider '%s' threw an exception: %s"
+                        .formatted(headerProvider.getClass().getSimpleName(), e.getMessage());
+                throw new DestinationAccessException(err, e);
+            }
+        }
+        return result;
+    }
+
+    @Nonnull
+    @Override
+    public Option<String> getTlsVersion()
+    {
+        return Option.none();
+    }
+
+    @Nonnull
+    @Override
+    public Option<ProxyConfiguration> getProxyConfiguration()
+    {
+        return Option.none();
+    }
+
+    @Nonnull
+    @Override
+    public Option<KeyStore> getKeyStore()
+    {
+        return Option.none();
+    }
+
+    @Nonnull
+    @Override
+    public Option<String> getKeyStorePassword()
+    {
+        return Option.none();
+    }
+
+    @Override
+    public boolean isTrustingAllCertificates()
+    {
+        return true;
```

Comment:
```suggestion
        return false;
```

I think this should default to false? In the (admittedly unlikely) case the transparent proxy is exposed under a `https` URL, we should not trust all certificates

Reply:
Done